### PR TITLE
feat: adding remote template expansion and validation of local .vela.yml files

### PR DESCRIPTION
**Pull request by GregoryDosh**
As part of https://github.com/go-vela/server/pull/448, this aims to add a new boolean flag to the CLI to support a local .vela.yml file to be supplied and sent off to the remote Vela server for template expansion and validation.

This is pretty rough as the control flow was hacked in. Could use some suggestions and naming changes. This is NOT ready for a merge just yet.

It requires some small changes to https://github.com/go-vela/sdk-go/pull/108 as well. Mostly just to add in a new service call to hit the different server endpoint. So for now the go.mod & go.sum are updated to point at the temporary dependency but that'd be removed before merging this in.

### CODE REVIEW

**Comment by KellyMerrick** [comment]
Has this effort been covered with https://github.com/go-vela/cli/pull/248 being merged in? Thanks!

**Comment by GregoryDosh** [comment]
I'm going to just copy & paste my related comment here: https://github.com/go-vela/server/pull/448#issuecomment-956511542
tldr Let's close this for now and reevaluate if this is needed in the future.
